src/MSALWrapper/AuthFlow/AuthFlowResult.cs

```diff
@@ -11,11 +11,19 @@ namespace Microsoft.Authentication.MSALWrapper
     /// </summary>
     public class AuthFlowResult
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthFlowResult"/> class with a null TokenResult and empty error list.
+        /// </summary>
+        public AuthFlowResult()
+            : this(null, null)
+        {
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="AuthFlowResult"/> class.
         /// </summary>
         /// <param name="tokenResult">A <see cref="MSALWrapper.TokenResult"/>.</param>
-        /// <param name="errors">A list of errors encountered while getting (or failing to get) the given token result.</param>
+        /// <param name="errors">A list of errors encountered while getting (or failing to get) the given token result. Will initialize a new empty List if null is given.</param>

```

Comment:
Why initialize a new empty `List` if `null` is given instead of just requiring a new empty `List` be given? Could the compiler also save us from `null` here?

Reply:
`null` is a valid input for the compiler. This code chunk was suggested by @kyle-rader so that anytime we have null, we initialize the errors as an empty list.

Reply:
Yeah, I think this is actually the same situation I was describing above where we should be able to just rely on the compiler to save us from a `null`. I forgot that you have to turn that on project-wide and couldn't just do it on a per-namespace basis. Another thing that should get noted as a thing that would be resolved by disabling nulls.

Reply:
I don't think forcing the caller to also make a new list should be the default path. But I also didn't like having to pass nulls (passing a null is a code smell to me that the signature you want doesn't exist yet).

Reply:
Wouldn't a variadic function signature be a more normal choice then? Only call with one argument if that's the one you have?